homeassistant/components/zwave_js/helpers.py

```diff
@@ -344,20 +345,59 @@ def async_get_nodes_from_area_id(
         }
     )
     # Add devices in an area that are Z-Wave JS devices
-    for device in dr.async_entries_for_area(dev_reg, area_id):
-        if next(
-            (
-                config_entry_id
-                for config_entry_id in device.config_entries
-                if cast(
+    nodes.update(
+        async_get_node_from_device_id(hass, device.id, dev_reg)
+        for device in dr.async_entries_for_area(dev_reg, area_id)
+        if any(
+            cast(
```

Comment:
This cast is unsafe as it is not handling the case where the entry_id is not present. In order to write safely this each entry should be checked whether is present in the entries before getting the value, making then the cast unnecessary.